Episode 16.2 - Friday Memorial Dungeon (#7300)
rathena/rathena: npc/re/instances/FridayDungeon.txt

		}
	end;
}

//Instance
1@md_gef,185,81	monster	Shining Plant	1083,1,1800000,900000

1@md_gef,110,129,6	script	Marry Jay#0_1	4_F_JOB_BLACKSMITH,{

	if (is_party_leader() == true) {
		mes "[Marry Jay]";
		mes "Before we start setting up the environment here, we'll give you the chance to choose an appropriate difficulty level.";
		mes "I could make my own choice, but each person handles it differently based on their abilities.";
		next;
		switch(select("Cancel:LV130 ~ 199:LV200+ ")) {
			case 1:
				close;
			//MinLevel, Lich Lord, Nightmare, Jakk, Ghoul, Drainliar
			case 2:
				setarray 'mob,130,3658,3660,3662,3664,3666; break;
			case 3:
				if(BaseLevel < 200){
				mes "[Marry Jay]";
					mes "You can't choose this difficulty, because your base level is below 200.";
					close;
				}
				'hard = 1;
				setarray 'mob,200,3659,3661,3663,3665,3667; break;
		}
		if(BaseLevel < 'mob[0]){
			mes "[Marry Jay]";
			mes "Your base level is too low to select this, you need to be level "+'mob[0]+" or above";
			close;
		}
		close2;
		npctalk "Now, this place is about to become a hunting ground. I'll get out and close the front door.";
		sleep 4000;
		npctalk "Good Luck~";
		sleep 3000;

> 		monster 'map$,0,0,"Nightmare",'mob[2],20,instance_npcname(strnpcinfo(0))+"::OnNightmareDead";
> 		monster 'map$,0,0,"Jakk",'mob[3],30,instance_npcname(strnpcinfo(0))+"::OnJakkDead";
> 		monster 'map$,0,0,"Ghoul",'mob[4],30,instance_npcname(strnpcinfo(0))+"::OnGhoulDead";
> 		monster 'map$,0,0,"Drainliar",'mob[5],20,instance_npcname(strnpcinfo(0))+"::OnDrainliarDead";
> 		hideonnpc instance_npcname(strnpcinfo(0));
> 		enablenpc instance_npcname("The Stranger#cadaver");
> 		enablenpc instance_npcname("Bizzare Sculpture#boss");

Reviewer: to prevent multiple spawning if the leader retalk within the time limit
```suggestion
		if ('step == 0) {
			'step = 1;
			monster 'map$,0,0,"Nightmare",'mob[2],20,instance_npcname(strnpcinfo(0))+"::OnNightmareDead";
			monster 'map$,0,0,"Jakk",'mob[3],30,instance_npcname(strnpcinfo(0))+"::OnJakkDead";
			monster 'map$,0,0,"Ghoul",'mob[4],30,instance_npcname(strnpcinfo(0))+"::OnGhoulDead";
			monster 'map$,0,0,"Drainliar",'mob[5],20,instance_npcname(strnpcinfo(0))+"::OnDrainliarDead";
			hideonnpc instance_npcname(strnpcinfo(0));
			enablenpc instance_npcname("The Stranger#cadaver");
			enablenpc instance_npcname("Bizzare Sculpture#boss");
		}
```

	}
	end;

OnNightmareDead:
	callsub S_ChkTreasure;
	sleep 10000;
	monster 'map$,0,0,"Nightmare",'mob[2],1,instance_npcname(strnpcinfo(0))+"::OnNightmareDead";
	end;
OnJakkDead:
	callsub S_ChkTreasure;
	sleep 10000;
	monster 'map$,0,0,"Jakk",'mob[3],1,instance_npcname(strnpcinfo(0))+"::OnJakkDead";
	end;
OnGhoulDead:
	callsub S_ChkTreasure;
	sleep 10000;
	monster 'map$,0,0,"Ghoul",'mob[4],1,instance_npcname(strnpcinfo(0))+"::OnGhoulDead";
	end;
OnDrainliarDead:
	callsub S_ChkTreasure;
	sleep 10000;
	monster 'map$,0,0,"Drainliar",'mob[5],1,instance_npcname(strnpcinfo(0))+"::OnDrainliarDead";
	end;

// Treasure Chest Spawn
S_ChkTreasure:
	'ct++;
	if ('ct == 100) {
		'rand = rand(1,4);
		'chest1 = 'rand;	//remember in case someone kills 100 enemies within 3 minutes
		enablenpc instance_npcname("#fd_box"+'chest1);
		instance_announce -1, "A treasure chest appeared somewhere in the dungeon. It'll disappear after a while, so let's find it.",bc_map,"0xff5500";

> 		sleep 180000;	//3min before chest disappears

Reviewer: The monster's respawn is delayed by 3min + 10s in this case. You may use something like donpcevent for example to call the treasure to not break the respawn.

irowiki states that `The treasure never disappears.` btw https://irowiki.org/wiki/Friday

Reviewer 2: @Atemo but on kRo chest dissapear after N time.

Author: All mechanics +text (translated) are fully based on kRO. They do de-spawn after 3 minutes, i confirmed it.
Can't remember about the monster spawn, i worked on this quite a while ago. Could be intended or a mistake on my part.

		instance_announce -1, "A treasure chest appeared somewhere in the dungeon. It'll disappear after a while, so let's find it.",bc_map,"0xff5500";
		sleep 180000;	//3min before chest disappears
		disablenpc instance_npcname("#fd_box"+'chest2);
	}
	if ('ct == 300) {
		'rand++;
		if ('rand >4) 'rand = 1;
		'chest3 = 'rand;	//remember in case someone kills 100 enemies within 3 minutes
		enablenpc instance_npcname("#fd_box"+'chest3);
		instance_announce -1, "A treasure chest appeared somewhere in the dungeon. It'll disappear after a while, so let's find it.",bc_map,"0xff5500";
		sleep 180000;	//3min before chest disappears
		disablenpc instance_npcname("#fd_box"+'chest3);
	}
	if ('ct == 400) {
		'rand++;
		if ('rand >4) 'rand = 1;
		'chest4 = 'rand;	//remember in case someone kills 100 enemies within 3 minutes
		enablenpc instance_npcname("#fd_box"+'chest4);
		instance_announce -1, "A treasure chest appeared somewhere in the dungeon. It'll disappear after a while, so let's find it.",bc_map,"0xff5500";
		sleep 180000;	//3min before chest disappears
		disablenpc instance_npcname("#fd_box"+'chest4);
	}
	return;

OnInstanceInit:
	'ct = 0;
	'freestones = 0;
	'boss_actv = 0;

> 	'map$ = instance_mapname(strnpcinfo(4));

Reviewer: ```suggestion
	'map$ = instance_mapname(strnpcinfo(4));
	'step = 0;
```

	case 2:
		mes "Looking at a hastily scribbled note in his pocket, perhaps the adventurer's name was Walter.";
		next;
		mes "[Adventurer's Log]";
		mes "My name is Walter. For the adventurers who will visit this place in the future, I'm leaving this so that you don't make the same mistake again.";
		next;
		mes "[Adventurer's Log]";
		mes "On the first day, I curiously put my hand on the slab of a strange Sculpture nearby.";
		mes "...and I awokened something that I should not have.";
		next;
		mes "[Adventurer's Log]";
		mes "Right now the revived ancient king is wandering in front of me.";
		mes "Perhaps he has no memories of his previous life.";
		mes "One thing is certain, he will kill me.";
		next;
		mes "[Adventurer's Log]";
		mes "To the adventurers who will visit this place in the future, I hope you do not make the same mistake as me.";
		mes "If you do see a suspicious slab, do not show any interest and pass by it.";
		next;
		mes "[Adventurer's Log]";
		mes "Again, never touch the slab. Whatever temptation you may have.";
		if('freestones==0) {
			next;
			mes "Several craft stones fall from his body. They were probably collected while exploring this place.";
			close2;

> 			if ( 'hard == 1 )
> 				.@num = rand(6+getmapusers('map$),8+getmapusers('map$));	// Reward amount scales with player count
> 			else
> 				.@num = rand(4+getmapusers('map$),6+getmapusers('map$));	// Reward amount scales with player count
> 				for ( .@i = 0; .@i < .@num; .@i++ )
> 					makeitem 25235,1, 'map$, rand(181,183), rand(221,224);
> 			'freestones=1;

Reviewer: Anyone can talk to the NPC. 'the freestones variable must be rechecked in the case that several st are running
```suggestion
			if('freestones==0) {
				if ( 'hard == 1 )
					.@num = rand(6+getmapusers('map$),8+getmapusers('map$));	// Reward amount scales with player count
				else
					.@num = rand(4+getmapusers('map$),6+getmapusers('map$));	// Reward amount scales with player count
					for ( .@i = 0; .@i < .@num; .@i++ )
						makeitem 25235,1, 'map$, rand(181,183), rand(221,224);
				'freestones=1;
			}
```

		} else
			close;
	}
	end;

OnInstanceInit:
	disablenpc();
	end;
}

1@md_gef,199,73,0	script	Bizzare Sculpture#boss	CLEAR_NPC,{
	if('ct < 100) {
		mes "It is a strange sculpture with a lot of dust piled up. I don't see anything suspicious.";
		close;
	}
	if('boss_actv >= 1) {
		mes "Craft Stones have been melted by a powerful force into the sockets.";
		mes "Seems like the sockets can't be used for a while.";
		close;
	}
	mes "At some point, the accumulated dust scatters, and 10 hollow holes are visible on the slab under the sculpture.";
	next;
	mes "Each of the 10 hollow sockets seem to perfectly fit a craft stone.";
	next;
	switch(select("Ignore:Try inserting craft stones")) {
		case 1:
			close;
		case 2:
			if(countitem(25235) >= 10) {
				mes "When you insert a craft stone into a hole, it emits a bright light and melts.";
				next;
				delitem 25235,10;
				specialeffect EF_ANGELUS;
				.@rand = rand(1,10);

> 				if(.@rand==10) {
> 					mes "Something seems to have changed in the air around me.";
> 					instance_announce -1, "Lord of the Dead: Who Awakened Me? I will meet you in person and ask you why.",bc_map,"0xFFFF00";
> 					monster 'map$,210,110,"Lich King",'mob[1],1;
> 					'boss_actv = 1;
> 				} else {
> 					mes "But nothing else happened.";
> 					close2;
> 					disablenpc();	// Disable NPC for 5 minutes when MVP didn't spawn.
> 					sleep 300000;
> 					enablenpc();
> 					end;

Reviewer: ```suggestion
				if(.@rand==10) {
					mes "Something seems to have changed in the air around me.";
					instance_announce -1, "Lord of the Dead: Who Awakened Me? I will meet you in person and ask you why.",bc_map,"0xFFFF00";
					if ('boss_actv == 0) {
						monster 'map$,210,110,"Lich King",'mob[1],1;
						'boss_actv = 1;
					}
				} else {
					mes "But nothing else happened.";
					close2;
					if ('boss_actv == 0) {
						disablenpc();	// Disable NPC for 5 minutes when MVP didn't spawn.
						sleep 300000;
						enablenpc();
					}
					end;
```

			else if (.@i < 175) .@enchant = 4722; //DEX+3
			else if (.@i < 225) .@enchant = 4723; //DEX+4
			else if (.@i < 235) .@enchant = 4724; //DEX+5
			else if (.@i < 245) .@enchant = 4725; //DEX+6
			else if (.@i < 250) .@enchant = 4726; //DEX+7
			else if (.@i < 300) .@enchant = 4752; //LUK+3
			else if (.@i < 350) .@enchant = 4753; //LUK+4
			else if (.@i < 360) .@enchant = 4754; //LUK+5
			else if (.@i < 370) .@enchant = 4755; //LUK+6
			else .@enchant = 4756; //LUK+7
			break;
		}
		return .@enchant;
	}

OnInit:
	.success_royal = 80;
	.success_normal = 20;
	.list$ = ",28483,2601,2602,2603,2605,2607,2608,2609,2610,2611,2612,2613,2614,2615,2616,2617,2618,2619,2620,2621,2622,2623,2625,2626,2627,2628,2636,2637,2638,2639,2640,2641,2645,2648,2649,2650,2651,2652,2653,2654,2655,2656,2658,2663,2664,2665,2666,2667,2671,2680,2692,2700,2701,2702,2703,2712,2714,2715,2716,2718,2719,2720,2721,2722,2723,2724,2725,2726,2727,2728,2729,2730,2731,2732,2736,2737,2743,2744,2745,2746,2747,2748,2749,2766,2767,2768,2769,2770,2771,2772,2773,2774,2777,2778,2779,2780,2781,2783,2784,2787,2788,2789,2790,2794,2795,2800,2801,2802,2803,2804,2805,2807,2808,2809,2810,2826,2827,2843,2853,2854,2881,2956,";

> 	end;

Reviewer: Definitely not fan of pre save the variables but ok

							close;
						case IE_NOINSTANCE:
							mes "The Friday Memorial Dungeon does not exist.";
							mes "Your party leader has not yet created the Memorial dungeon.";
							close;
						case IE_OTHER:
							mes "An unknown error occurred.";
							close;
						case IE_OK:
							mes "[Marry Jay]";
							mes "Off you go!";
							mapannounce "gef_tower", "" + strcharinfo(0) + " of the party, " + getpartyname( getcharid(1) ) + ", is entering the " + .@md_name$ + ".", bc_map, 0xFF99;
							if(isbegin_quest(12379)<1) setquest 12379;
							end;
					}
					break;
				case 2:
					mes "[Marry Jay]";
					mes "You have not prepared?";
					mes "Can't really help it I guess";
					close;
			}
		}
	end;
}

//Instance
1@md_gef,185,81	monster	Shining Plant	1083,1,1800000,900000

1@md_gef,110,129,6	script	Marry Jay#0_1	4_F_JOB_BLACKSMITH,{

	if (is_party_leader() == true) {
		mes "[Marry Jay]";
		mes "Before we start setting up the environment here, we'll give you the chance to choose an appropriate difficulty level.";
		mes "I could make my own choice, but each person handles it differently based on their abilities.";
		next;

> 		switch(select("Cancel:LV130 ~ 199:LV200+ ")) {

Reviewer 3: ```suggestion
		switch(select("Cancel:LV130 ~ 199:LV200+")) {
```

			case 1:
				close;
			//MinLevel, Lich Lord, Nightmare, Jakk, Ghoul, Drainliar
			case 2:
				setarray 'mob,130,3658,3660,3662,3664,3666; break;
			case 3:
				if(BaseLevel < 200){
				mes "[Marry Jay]";
					mes "You can't choose this difficulty, because your base level is below 200.";
					close;
				}
				'hard = 1;
				setarray 'mob,200,3659,3661,3663,3665,3667; break;
		}
		if(BaseLevel < 'mob[0]){
			mes "[Marry Jay]";

> 			mes "Your base level is too low to select this, you need to be level "+'mob[0]+" or above";

Reviewer 3: ```suggestion
			mes "Your base level is too low to select this, you need to be level "+'mob[0]+" or above.";
```

		close;
	}
	if (MaxWeight - Weight < 10000) {
		mes "Can't continue because you have too many heavy objects. Reduce your total weight by clearing your inventory.";
		close;
	}
	mes "[Amateur Collector]";
	mes "If you bring me a <ITEM>"+getitemname(25235)+"<INFO>25235</INFO></ITEM>, I can enchant a wonderful effect on your accessory!";
	next;
	switch(select("What can you do with the Crafting Stone?:Accessory Enchantment:Reset Enchant")) {
		case 1:
			mes "[Amateur Collector]";
			mes "I'm a trade expensive stones from the dungeon in exchange for my craftsmanship. I'm still a beginner, but my skills are passable.";
			mes "Please take care of me.";
			next;
			switch(select("I've heard this before.:I'm listening.")) {
				case 1:
					mes "[Amateur Collector]";
					mes "Is that so?";
					close;
				case 2:
					mes "[Amateur Collector]";
					mes "By using the Craft Stone, I can enchant one of your accessories with abilities.";
					next;
					mes "[Amateur Collector]";
					mes "I need 10 x <ITEM>"+getitemname(25235)+"<INFO>25235</INFO></ITEM> and 100,000 Zeny.";
					mes "There is no failure in the enchantment process.";
					next;
					mes "[Amateur Collector]";

> 					mes "However ^0000FF if the accessory has a random option, the random option will be removed after enchantment.^000000";

Reviewer 3: ```suggestion
					mes "However ^0000FFif the accessory has a random option, the random option will be removed after enchantment.^000000";
```

					mes "Are you sure to enhance the <ITEM>"+getitemname(.@equip_id)+"<INFO>"+.@equip_id+"</INFO></ITEM>?";
					mes "^0000FFThere is a no chance of failure^000000.";
					next;
					if(select("Yes, please proceed:Cancel")==2) close;
				}
				if(countitem(25235)<10) {
					mes "[Amateur Collector]";
					mes "Please bring me 10 x <ITEM>"+getitemname(25235)+"<INFO>25235</INFO></ITEM>";
					close;
				}
				if(Zeny < 100000) {
					mes "[Amateur Collector]";
					mes "I need 100,000 zeny!";
					close;
				}
				delitem 25235,10; //Catalyst
				Zeny -= 100000;
				delequip .@slot;
				specialeffect2 EF_REPAIRWEAPON;
				mes "[Amateur Collector]";
				mes "The <ITEM>"+getitemname(.@equip_id)+"<INFO>"+.@equip_id+"</INFO></ITEM> has been enchanted!!";
				getitem2 .@equip_id,1,1,.@equip_refine,0,.@equip_card[0],.@equip_card[1],.@equip_card[2],.@equip_card[3];
			//Normal Accessory Enchantment
			} else {
				if(.@equip_card[3] != 0) {
					mes "[Amateur Collector]";
					mes "The <ITEM>"+getitemname(.@equip_id)+"<INFO>"+.@equip_id+"</INFO></ITEM> already has an enchant.";
					mes "I cannot work on this any further.";
					close;
				}
				mes "[Amateur Collector]";

> 				mes "Are you sure to enhance the <ITEM>"+getitemname(.@equip_id)+"<INFO>"+.@equip_id+"</INFO></ITEM>?";

Reviewer 3: ```suggestion
				mes "Are you sure you want to enhance the <ITEM>"+getitemname(.@equip_id)+"<INFO>"+.@equip_id+"</INFO></ITEM>?";
```

				mes "^0000FFThere is a no chance of failure^000000.";
				next;
				if(select("Yes, please proceed:Cancel")==2) close;
				if(countitem(25235)<10) {
					mes "[Amateur Collector]";
					mes "Please bring me 10 x <ITEM>"+getitemname(25235)+"<INFO>25235</INFO></ITEM>";
					close;
				}
				if(Zeny < 100000) {
					mes "[Amateur Collector]";
					mes "I need 100,000 zeny!";
					close;
				}
				delitem 25235,10; //Catalyst
				Zeny -= 100000;
				delequip .@slot;
				.@enchant = enchant(1);
				specialeffect2 EF_REPAIRWEAPON;
				mes "[Amateur Collector]";

> 				mes "The <ITEM>"+getitemname(.@equip_id)+"<INFO>"+.@equip_id+"</INFO></ITEM> has been enhanted with <ITEM>"+getitemname(.@enchant)+"<INFO>"+.@enchant+"</INFO></ITEM>.";

Reviewer 3: ```suggestion
				mes "The <ITEM>"+getitemname(.@equip_id)+"<INFO>"+.@equip_id+"</INFO></ITEM> has been enchanted with <ITEM>"+getitemname(.@enchant)+"<INFO>"+.@enchant+"</INFO></ITEM>.";
```
